From b266300d4e089e1f281305d99c7f6b3bbc94dae9 Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Tue, 26 Mar 2024 18:01:52 +0530
Subject: [PATCH] wifi: ath12k: disable ppdu stats req based on stats_disable
 debugfs

During performence measurement, ppdu stats request is disabled through
stats_disable debugfs. But, its enabled again during vdev up. So,
enable this stats request based on stats_disable debugfs check.

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index de14a8b..cb944ed 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -10185,11 +10185,13 @@ int ath12k_mac_radio_start(struct ath12k *ar)
 		goto err;
 	}
 
-	ret = ath12k_dp_tx_htt_h2t_ppdu_stats_req(ar,
+	if (!ab->stats_disable) {
+		ret = ath12k_dp_tx_htt_h2t_ppdu_stats_req(ar,
 						  HTT_PPDU_STATS_TAG_DEFAULT);
-	if (ret) {
-		ath12k_err(ab, "failed to req ppdu stats: %d\n", ret);
-		goto err;
+		if (ret) {
+			ath12k_err(ab, "failed to req ppdu stats: %d\n", ret);
+			goto err;
+		}
 	}
 
 	ret = ath12k_wmi_pdev_set_param(ar, WMI_PDEV_PARAM_MESH_MCAST_ENABLE,
-- 
2.17.1

